entity-framework/core/what-is-new/ef-core-3.0/index.md

```diff
@@ -77,7 +77,7 @@ var orders =
   where o.Status == OrderStatus.Pending
   select o;
 
-await foreach(var o in orders)
+await foreach(var o in orders.AsAsyncEnumerable())
```

Comment:
Why isn't this pattern matched by the compiler?

Reply:
Orders is of type `IQueryable<Order>` IQueryable implements IEnumerable (which compiler matches for foreach). IQueryable does not implement IAsyncEnumerable. :

Reply:
But there’s clearly an extension method that can handle it. The compiler uses extension methods in plenty of other places...

Reply:
lol. File an issue with Roslyn?

Reply:
Not a crazy idea... Actually, it should match if there was a GetAsyncEnumerator<T\> method that matches!

Reply:
I tried this and I confirmed that it doesn't work :cry:. So I was about to file an issue for it, but i found that @jcouv already filed it at https://github.com/dotnet/roslyn/issues/32289.

I am going to add a comment, but feel free to upvote.